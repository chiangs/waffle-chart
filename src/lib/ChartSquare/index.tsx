import React from 'react';
import type { GridItemProps } from '../__types';

type Props = {
    gridItemProps: GridItemProps;
    isAnimatedFill: boolean;
    isZeros: boolean;
    clickHandler: (props: GridItemProps) => GridItemProps;
};

const NAME_COMPONENT = `waffle-chart-square`;

const ChartSquare: React.FC<Props> = ({
    gridItemProps,
    isAnimatedFill,
    isZeros,
    clickHandler,
}: Props) => {
    const classes = [NAME_COMPONENT];
    classes.push(gridItemProps?.identifier);
    if (isAnimatedFill) classes.push(`animate-fill`);
    if (isZeros) classes.push('zeroed');
    // Fallback style
    const fallbackStyle = {
        background: `var(--bg-fallback-${gridItemProps?.identifier})`,
    };
    return (
        <div
            data-testid={NAME_COMPONENT}
            className={classes.join(' ')}
            style={gridItemProps?.style || fallbackStyle}
            key={gridItemProps?.index}
            title={`square${gridItemProps?.index}`}
            onClick={() => clickHandler(gridItemProps)}></div>
    );
};

export default ChartSquare;
