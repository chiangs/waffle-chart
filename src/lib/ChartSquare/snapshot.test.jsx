import React from 'react';
import renderer from 'react-test-renderer';
import { describe, expect, test } from 'vitest';
import ChartSquare from '.';

describe('Chart', () => {
    test('Chart component renders correctly', () => {
        const props = {
            props: [],
            isAnimatedFill: true,
            isZeros: true,
            clickHandler: () => null,
        };
        const component = renderer.create(<ChartSquare {...props} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
