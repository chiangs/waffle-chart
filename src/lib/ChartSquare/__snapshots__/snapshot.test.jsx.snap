// Vitest Snapshot v1

exports[`Chart > Chart component renders correctly 1`] = `
<div
  className="waffle-chart-square  animate-fill zeroed"
  data-testid="waffle-chart-square"
  onClick={[Function]}
  style={
    {
      "background": "var(--bg-fallback-undefined)",
    }
  }
  title="squareundefined"
/>
`;
